### PR TITLE
Code and Docs improvements

### PR DESCRIPTION
**Pull request by kevinpapst**
- Fixed several deprecations
- Fixed PHPStan issues
- and adapted the README to better reflect the Version 2, especially the fact that the user work hours are now configured in the core settings of the user and that the Custom-Fields plugin is not required anymore

### CODE REVIEW

**Comment by kevinpapst** [comment]
Please double-check the README, it contained infos that are no longer true for v2 (the custom-fields part).

I would recommend to add a prominent link that explains, that users of v1 MUST read the README from your other branch!

File: README.md [hunk 1]
```diff
@@ -8,18 +8,22 @@ Here is a short live demo (for Kimai2 in version 1):
...
 ## Requirements
 
-- Requires Kimai 2, V1.16.10 - V1.30.11 (for ApprovalBundle in Version 1.x as in `bundle_for_kimai_v1`)
-- Required Kimai 2, >= Version 2.5 (for ApprovalBundle in Version 2.x as in default `main`) 
+- Required Kimai >= 2.5 (for ApprovalBundle in Version 2.x in branch `main`) 
+- Requires Kimai >= 1.16.10 (for ApprovalBundle in Version 1.x in branch `bundle_for_kimai_v1`)
 
-Optional but recommended:
-- LockdownPerUser plugin ([GitHub](https://github.com/kevinpapst/LockdownPerUserBundle)) - without the lockdown functionality will not work
-- MetaFields plugin - optional setting of working hours per day per user unless specified in ApprovalBundle "Settings workday"
```
Comment (KatjaGlassConsulting) [inline README.md:22]: Hi Kevin, just for clarification, the MetaFields plugin is no longer required, as this had been used for the expected working time which is now fixed available in Kimai V2 an accessible by user->getWorkHoursMonday etc.

I guess this still needs to be documented as per default these times (or in V1 the time entered through meta-fields) are used when no bundle-specific setting is done. The bundle specific setting should only be applied when times changes.

**Comment by KatjaGlassConsulting** [comment]
I prefere to have just one readme synced for both version - I will make some updates though, also in V1 the metaFields is optional.

File: README.md (hunk 1 above)
Comment (kevinpapst) [reply README.md:22]: Correct, these settings are now available in Kimai-Core.
Users of the WorkingContract plugin have their previous settings automatically migrated.

But users of the Custom-Field bundle and the ApprovalBundle alone need to configure the expected working times again. I haven't thought about documenting that.

Most projects I know use the different branches for the different docs, so I didn't anticipate that. You shouldn't merge so fast, I could have adapted my changed 😄 
When you want to have one README for both versions, we need to revert some of my change and probably put the infos under different headings (e.g. `Requirements for v1`, `Requirements for v2`, `Configuration for v1`, `Configuration for v2`)?

**Pull request merged**